src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -6067,7 +6091,7 @@ class CreateDatabaseRoleStatementSegment(BaseSegment):
             "IfNotExistsGrammar",
             optional=True,
         ),
-        Ref("ObjectReferenceSegment"),
+        Ref("DatabaseRoleReferenceSegment"),
```

Comment:
This was a small improvement on my previous PR https://github.com/sqlfluff/sqlfluff/pull/5475. If SQLFluff very strictly follows the 1 PR = 1 change rule, I can back this change out.

Reply:
Nope, we can group together related changes where it makes sense. 👍